### PR TITLE
New test for verify pvc ownership fix.

### PR DESCRIPTION
**Pull request by Rohit-PX**
Signed-off-by: Rohit-PX <rohit@portworx.com>

**What type of PR is this?**
integration-test

**What this PR does / why we need it**:
We need this PR to automate integration test for this issue:
https://github.com/libopenstorage/stork/issues/291

**Does this PR change a user-facing CRD or CLI?**:
No

**Is a release note needed?**:
No

```

**Does this change need to be cherry-picked to a release branch?**:
No

### CODE REVIEW

File: test/integration_test/extender_test.go [hunk 1]
```diff
@@ -146,3 +148,52 @@ func driverNodeErrorTest(t *testing.T) {
...
+				}
+			}
+		}
+	}
+
+	for _, spec := range ctx.App.SpecList {
+		if obj, ok := spec.(*vi.PersistentVolumeClaim); ok {
+			updatePVC, _ := k8sOps.GetPersistentVolumeClaim(obj.Name, obj.Namespace)
+			if annotationValue, hasKey := updatePVC.Annotations["storage-provisioner"]; hasKey {
+				delete(updatePVC, "storage-provisioner")
```
Comment (disrani-px) [inline test/integration_test/extender_test.go]: Need to call UpdatePersistentVolumeClaim() after updating the annotation

File: test/integration_test/extender_test.go [hunk 2]
```diff
@@ -6,16 +6,21 @@ import (
...
 )
 
 func testExtender(t *testing.T) {
-	t.Run("noPVCTest", noPVCTest)
-	t.Run("singlePVCTest", singlePVCTest)
-	t.Run("statefulsetTest", statefulsetTest)
-	t.Run("multiplePVCTest", multiplePVCTest)
-	t.Run("driverNodeErrorTest", driverNodeErrorTest)
+	t.Run("pvcOwnershipTest", pvcOwnershipTest)
+	//t.Run("noPVCTest", noPVCTest)
```
Comment (disrani-px) [inline test/integration_test/extender_test.go]: Uncomment these

File: test/integration_test/extender_test.go [hunk 3]
```diff
@@ -146,3 +151,65 @@ func driverNodeErrorTest(t *testing.T) {
...
+
+	for _, spec := range ctxs[0].App.SpecList {
+		if obj, ok := spec.(*storage_api.StorageClass); ok {
+			if err := k8s.Instance().DeleteStorageClass(obj.Name); err != nil {
+				require.NoError(t, err, "Error deleting storage class for mysql.")
+			}
+		}
+		if obj, ok := spec.(*v1.PersistentVolumeClaim); ok {
+			updatePVC, _ := k8s.Instance().GetPersistentVolumeClaim(obj.Name, obj.Namespace)
+			if _, hasKey := updatePVC.Annotations["storage-provisioner"]; hasKey {
```
Comment (disrani-px) [inline test/integration_test/extender_test.go]: The annotation is `volume.beta.kubernetes.io/storage-provisioner`. Define a const for that and use that here

File: test/integration_test/extender_test.go [hunk 4]
```diff
@@ -146,3 +151,65 @@ func driverNodeErrorTest(t *testing.T) {
...
+	var errUnscheduledPod bool
+	var depPods []v1.Pod
+	for _, spec := range ctxs[0].App.SpecList {
+		if obj, ok := spec.(*apps_api.Deployment); ok {
+			if obj.Name == "mysql" {
+				depPods, err = k8s.Instance().GetDeploymentPods(obj)
+				require.NoError(t, err, "Error getting pods for deployment ,mysql.")
+			}
+		}
+		for _, pod := range depPods {
```
Comment (disrani-px) [inline test/integration_test/extender_test.go]: Put this within the `if` above. It only needs to be done for the mysql deployment.

File: test/integration_test/extender_test.go [hunk 5]
```diff
@@ -146,3 +151,65 @@ func driverNodeErrorTest(t *testing.T) {
...
+			for _, cond := range pod.Status.Conditions {
+				if cond.Type == v1.PodScheduled && cond.Status == v1.ConditionFalse {
+					errUnscheduledPod = true
+				}
+			}
+		}
+	}
+	require.Equal(t, false, errUnscheduledPod, "Pod should not have been schedule.")
```
Comment (disrani-px) [inline test/integration_test/extender_test.go]: Expected is `true`, I think you didn't push your change.

File: test/integration_test/extender_test.go [hunk 6]
```diff
@@ -146,3 +151,65 @@ func driverNodeErrorTest(t *testing.T) {
...
+	verifyScheduledNode(t, scheduledNodes[0], volumeNames)
+
+	for _, spec := range ctxs[0].App.SpecList {
+		if obj, ok := spec.(*storage_api.StorageClass); ok {
+			if err := k8s.Instance().DeleteStorageClass(obj.Name); err != nil {
+				require.NoError(t, err, "Error deleting storage class for mysql.")
+			}
+		}
+		if obj, ok := spec.(*v1.PersistentVolumeClaim); ok {
+			updatePVC, _ := k8s.Instance().GetPersistentVolumeClaim(obj.Name, obj.Namespace)
```
Comment (disrani-px) [inline test/integration_test/extender_test.go]: Don't ignore error.

File: test/integration_test/extender_test.go [hunk 7]
```diff
@@ -146,3 +151,65 @@ func driverNodeErrorTest(t *testing.T) {
...
+	require.Equal(t, 1, len(scheduledNodes), "App should be scheduled on one node")
+
+	volumeNames := getVolumeNames(t, ctxs[0])
+	require.Equal(t, 1, len(volumeNames), "Should have only one volume")
+
+	verifyScheduledNode(t, scheduledNodes[0], volumeNames)
+
+	for _, spec := range ctxs[0].App.SpecList {
+		if obj, ok := spec.(*storage_api.StorageClass); ok {
+			if err := k8s.Instance().DeleteStorageClass(obj.Name); err != nil {
```
Comment (disrani-px) [inline test/integration_test/extender_test.go]: No need to use `if` to check for err. Just call `require.NoError()`. Same for the other calls below

File: test/integration_test/extender_test.go [hunk 8]
```diff
@@ -6,11 +6,20 @@ import (
...
+	"github.com/portworx/sched-ops/k8s"
 	"github.com/portworx/torpedo/drivers/scheduler"
 	"github.com/stretchr/testify/require"
+	apps_api "k8s.io/api/apps/v1beta2"
+	"k8s.io/api/core/v1"
+	storage_api "k8s.io/api/storage/v1"
+)
+
+const (
+	annotation_storage_provisioner = "storage-provisioner"
```
Comment (disrani-px) [inline test/integration_test/extender_test.go]: `annotationStorageProvisioner`

File: test/integration_test/extender_test.go (hunk 1 above)
Comment (Rohit-PX) [reply test/integration_test/extender_test.go]: Fixed.

File: test/integration_test/extender_test.go (hunk 6 above)
Comment (Rohit-PX) [reply test/integration_test/extender_test.go]: Fixed.

File: test/integration_test/extender_test.go (hunk 5 above)
Comment (Rohit-PX) [reply test/integration_test/extender_test.go]: Fixed.

File: test/integration_test/extender_test.go (hunk 4 above)
Comment (Rohit-PX) [reply test/integration_test/extender_test.go]: Fixed.

File: test/integration_test/extender_test.go (hunk 3 above)
Comment (disrani-px) [reply test/integration_test/extender_test.go]: The annotation in the const is incorrect

**Pull request merged**